pkg/httpmock/http.go

```diff
@@ -121,6 +121,30 @@ func setup() (*openapi.APIClient, *openapi.Configuration, *http.ServeMux, *httpt
 	return c, clientCfg, mux, server, port, server.Close
 }
 
+func MutatingResponse(filename string, callback func(count int, i []byte) ([]byte, error)) http.HandlerFunc {
+	count := 0
```

Comment:
You probably should use a mutex here, I think this will hinder parallel tests using this func.

Reply:
Probably yes, I'll add it in.

Reply:
How is the test failing? deadlock?

Reply:
yep, deadlock.